addons/hr_holidays/models/hr_employee.py

```diff
@@ -300,14 +300,15 @@ def get_special_days_data(self, date_start, date_end):
     @api.model
     def get_public_holidays_data(self, date_start, date_end):
         self = self._get_contextual_employee()
+        employee_tz = pytz.timezone(self._get_tz())
```

Comment:
```python
employee_tz = pytz.timezone(self._get_tz() if self else self.env.user.tz)
```

Reply:
```python
    def _get_tz(self):
        # Finds the first valid timezone in his tz, his work hours tz,
        #  the company calendar tz or UTC and returns it as a string
        if len(self) > 1:
            self.ensure_one()
        return self.tz or\
               self.resource_calendar_id.tz or\
               self.company_id.resource_calendar_id.tz or\
               self.env.company.resource_calendar_id.tz or\
               self.env.user.tz or\
               'UTC'
```

Reply:
the one beneath would also work if you want to centralize stuff which is at least a good effort...